storage/innobase/xtrabackup/src/ddl_tracker.cc

```diff
@@ -0,0 +1,1041 @@
+      list. for example: 1. t1.ibd is discovered
+                   2. t1.ibd renamed to t2.ibd
+                   3. t2.ibd is opened and loaded to cache to copy
+                   4. t1.ibd is missing now
+      so we should add t2.ibd to new_tables and skip .ren file so that we don't
+      try to rename t1.ibd to t2.idb where t1.ibd is missing   */
+    if (new_tables.find(space_id) != new_tables.end() ||
+        is_missing_table(old_table_name)) {
+      new_tables[space_id] = new_table_name;
+      continue;
+    }
+
+    /* Table not in the backup, nothing to rename, skip rename*/
+    if (tables_in_backup.find(space_id) == tables_in_backup.end()) {
+      continue;
+    }
+
+    backup_file_printf(
+        convert_file_name(space_id, old_table_name, EXT_REN).c_str(), "%s",
+        new_table_name.c_str());
+  }
+
+  for (auto table = new_tables.begin(); table != new_tables.end();) {
+    if (check_if_skip_table(table->second.c_str())) {
+      table = new_tables.erase(table);
+      continue;
+    }
+
+    if (fsp_is_system_tablespace(table->first)) {
+      // open system tablespace for recopying
+      srv_sys_space.shutdown();
+
+      srv_sys_space.set_space_id(TRX_SYS_SPACE);
+
+      /* Create the filespace flags. */
+      ulint fsp_flags =
+          fsp_flags_init(univ_page_size, false, false, false, false);
+      srv_sys_space.set_flags(fsp_flags);
+
+      srv_sys_space.set_name("innodb_system");
+      srv_sys_space.set_path(srv_data_home);
+
+      /* Supports raw devices */
+      if (!srv_sys_space.parse_params(innobase_data_file_path, true,
+                                      xtrabackup_prepare)) {
+        xb::error() << "Cannot parse system tablespace params";
+        ut_ad(0);
+        return;
+      }
+      dberr_t err;
+      page_no_t sum_of_new_sizes;
+      lsn_t flush_lsn;
+
+      std::this_thread::sleep_for(std::chrono::milliseconds(200));
+
+      err = srv_sys_space.check_file_spec(false, 0);
+
+      if (err != DB_SUCCESS) {
+        xb::error() << "could not find data files at the specified datadir";
+        ut_ad(0);
+        return;
+      }
+
+      err = srv_sys_space.open_or_create(false, false, &sum_of_new_sizes,
+                                         &flush_lsn);
+
+      if (err != DB_SUCCESS) {
+        xb::error() << "could not reopen system tablespace";
+        ut_ad(0);
+        return;
+      }
+
+    } else {
+      std::tie(err, std::ignore) = fil_open_for_xtrabackup(
+          table->second, table->second.substr(0, table->second.length() - 4));
+    }
+    table++;
+  }
+
+  // Create .del files for deleted undo tablespaces
+  for (auto &elem : deleted_undo_files) {
+    backup_file_printf(
+        convert_file_name(elem.second, elem.first, EXT_DEL).c_str(), "%s", "");
+  }
+
+  // Add new undo files to be recopied
+  for (auto &elem : new_undo_files) {
+    new_tables[elem.second] = elem.first;
+  }
+
+  if (new_tables.empty()) return;
```

Comment:
Lets indicate that there are no files are being copied by reduced lock. xb::info() msg please

Reply:
AB